docs/tutorials/building-your-first-bot.mdx

```diff
@@ -0,0 +1,342 @@
+    {
+        // Loads the bot prefab and spawns it into the scene
+        var bot = Instantiate(rgBotPrefab, Vector3.zero, Quaternion.identity);
+        bot.transform.position = botSpawnPoint.position;
+        return bot;
+    }
+    
+    void Start()
+    {
+        // Prepares the RG SDK for spawning bots
+        RGBotServerListener.GetInstance()?.StartGame();
+        RGBotServerListener.GetInstance()?.SpawnBots();
+    }
+
+    private void OnDestroy()
+    {
+        // Stops all bots and cleans up the RG SDK
+        RGBotServerListener.GetInstance()?.StopGame();
+    }
+
+}
+```
+
+Once this component is implemented, create an empty object within the scene, rename it to `RGBotSpawner` (just so we can find it later),
+and attach this `CharacterBotSpawnManager` component to that object. Position it within the scene wherever you want to spawn your bot 
+(**make sure to set the Y coordinate to 0 so your bot does not float!**), and drag `RGBotSpawner`'s own Transform into the 
+**Bot Spawn Point** entry of the RGBotSpawner's `CharacterBotSpawnManager` component. Finally, find the 
+**Player** prefab provided by the sample project (located within `Assets/FirstBotDemo/Demo/Prefabs/Player.prefab`), and drag 
+this into the **Rg Bot Prefab** entry. Regression Games now knows how to load and unload bots from your scene! Make sure to save the 
+scene.
+
+![Screenshot of the RGOverlayCanvas.](./img/building-your-first-bot/spawner_6.png)
+
+## Add state information to relevant GameObjects
+
+We are onto the final parts of the integration! We need to indicate the following within our game:
+
+* What game state our bots can see
+* What actions our bots can perform
+
+First, let's add the state information. Regression Games marks states for bot consumption using the 
+[`[RGState]` attribute](/integrating-with-unity/defining-states#3---rgstate-fieldpropertymethod-attribute).
+This attribute can be applied to any public method that returns a value or any variable within a `MonoBehaviour`, and
+it will automatically be sent to the bot.
+
+First, open the `Assets/FirstBotDemo/Demo/Scripts/Player/PlayerAttack.cs` file, and add the following method to the class, which
+will allow our bot to know when it is currently in an attack animation (make sure to import any missing imports, with
+`using RegressionGames;`):
+
+```csharp
+[RGState]
+public bool IsAttacking()
+{
+    return !string.IsNullOrEmpty(_attackAnimation) && _animator.GetBool(_attackAnimation);
+}
+```
+
+Then, within the `Assets/FirstBotDemo/Demo/Scripts/EnemyController.cs` file, add the following attributes to the existing `GetCurrentHp()`
+and `GetTotalHp()` methods:
+
+```csharp
+/*
+* Gets the enemies current hp
+*/
+[RGState("CurrentHealth")]
+public int GetCurrentHp()
+{
+    return _currentHp;
+}
+
+/*
+* Gets the enemy's total HP from its assigned info
+*/
+[RGState("MaxHealth")]
+public int GetTotalHp()
```

Comment:
I was hoping to avoid changing the original code, but maybe I can... I was about to respond to this saying we have examples in our RGState code, but I realize we don't seem to have examples in the reference for these difference options. We should likely add that at some point as well...

Reply:
@RG-nAmKcAz I couldn't find this in the docs, but does the field need to be public to be accessible with the new state approach? If you see my new code above on line 203, I added the attribute to a private field - I still need to test but figured I'd ask here.